app/src/main/java/com/andela/helpmebuy/fragments/TravelListFragment.java

```diff
@@ -54,122 +44,72 @@
         if (!isRunning) {
             homeCountryDetector.connect();
         }
     }
 
-    @Override
-    protected void onStop() {
-        homeCountryDetector.disconnect();
-        super.onStop();
-    }
-
-    private void addActionBar() {
-        toolbar = (Toolbar) findViewById(R.id.toolbar);
-        setSupportActionBar(toolbar);
-        ActionBar actionBar = getSupportActionBar();
-
-        if (actionBar != null) {
-            actionBar.setDisplayHomeAsUpEnabled(true);
-            actionBar.setHomeButtonEnabled(true);
-        }
-    }
-
-    private void loadComponents() {
-        parentLayout = (CoordinatorLayout) findViewById(R.id.parent_layout);
-        drawerLayout = (DrawerLayout) findViewById(R.id.home_activity_drawer_layout);
-
-        drawerToggle = new ActionBarDrawerToggle(this, drawerLayout, R.string.drawer_open, R.string.drawer_close);
-        drawerToggle.syncState();
-
-        navigationView = (NavigationView) findViewById(R.id.home_activity_navigation_view);
-        navigationView.setNavigationItemSelectedListener(this);
-
-        View drawerHeader = LayoutInflater.from(this).inflate(R.layout.home_activity_drawer_header, null);
-        navigationView.addHeaderView(drawerHeader);
-        usernameTextView = (TextView) drawerHeader.findViewById(R.id.user_name_text);
-        userEmailTextView = (TextView) drawerHeader.findViewById(R.id.user_email_text);
+    private void loadComponents(final View rootView) {
+        FloatingActionButton fab = (FloatingActionButton) rootView.findViewById(R.id.search_fab);
+        fab.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                changeLocation();
 
-        travellersView = (RecyclerView) findViewById(R.id.travellers_recycler_view);
-        travellersView.addItemDecoration(new ItemDivider(this));
+            }
+        });
+        travellersView = (RecyclerView) rootView.findViewById(R.id.travellers_recycler_view);
+        travellersView.addItemDecoration(new ItemDivider(context));
         registerForContextMenu(travellersView);
 
-        LinearLayoutManager layoutManager = new LinearLayoutManager(this);
+        LinearLayoutManager layoutManager = new LinearLayoutManager(context);
         travellersView.setLayoutManager(layoutManager);
         travels = new ArrayList<>();
-        adapter = new TravellersAdapter(this, travels);
+        adapter = new TravellersAdapter(context, travels);
         adapter.setCurrentTravelListener(this);
         travellersView.setAdapter(adapter);
-        setUserProfile(this);
-
-        notify = (TextView) findViewById(R.id.notify);
-        progressWheel = (ProgressWheel) findViewById(R.id.progress_wheel);
+        notify = (TextView) rootView.findViewById(R.id.notify);
+        progressWheel = (ProgressWheel) rootView.findViewById(R.id.progress_wheel);
 
         notify.setVisibility(View.INVISIBLE);
         progressWheel.spin();
```

Comment:
NITPICK: Why not let us be consistent across the app. If we want to maintain the spinners or just a message  just as we did for the contact(`no contacts`). I'd suggest we adopt the message method as to the spinners. This would be a better UX.